src/main/java/org/openmaptiles/layers/WaterName.java

```diff
@@ -176,20 +189,32 @@ public void process(Tables.OsmWaterPolygon element, FeatureCollector features) {
         Geometry centerlineGeometry = lakeCenterlines.get(element.source().id());
         FeatureCollector.Feature feature;
         int minzoom = 9;
+        String place = element.place();
+        String clazz;
+        if ("bay".equals(element.natural())) {
+          clazz = FieldValues.CLASS_BAY;
+        } else if ("sea".equals(place)) {
+          clazz = FieldValues.CLASS_SEA;
+        } else {
+          clazz = FieldValues.CLASS_LAKE;
+          minzoom = 3;
+        }
         if (centerlineGeometry != null) {
           // prefer lake centerline if it exists
           feature = features.geometry(LAYER_NAME, centerlineGeometry)
             .setMinPixelSizeBelowZoom(13, 6d * element.name().length());
         } else {
           // otherwise just use a label point inside the lake
           feature = features.pointOnSurface(LAYER_NAME);
-          Geometry geometry = element.source().worldGeometry();
-          double area = geometry.getArea();
-          minzoom = (int) Math.floor(20 - Math.log(area / WORLD_AREA_FOR_70K_SQUARE_METERS) / LOG2);
-          minzoom = Math.min(14, Math.max(9, minzoom));
+          double area = element.source().area();
+          if (place != null && SEA_OR_OCEAN_PLACE.contains(place)) {
+            minzoom = 0;
+          } else {
+            minzoom = areaToMinZoom(area);
+          }
```

Comment:
TODO for planetiler: "include a point at the the center of this polygon when the polygon's tile area is at least X pixels" seems like a common use-case that should be pushed down into planetiler.  Maybe reuse `.setMinPixelSize(...)` which currently doesn't have any meaning for points or a new feature method: `.whenFeatureSizeExceeds(...)`

Reply:
back-link: https://github.com/onthegomap/planetiler/issues/716

This looks nicer, hence I'll adjust the PR using the new core feature.